From 6f79f210c690c31bce632570bf703a2360af6827 Mon Sep 17 00:00:00 2001
From: Clemens Lang <neverpanic@gmail.com>
Date: Sun, 6 Mar 2016 21:28:56 +0100
Subject: [PATCH 1/2] Revert "perlasm/x86_64-xlate.pl: pass pure constants
 verbatim."

This reverts commit fd7dc201d3b9d43972de6a0e659f7ef6421c99cc.

This commit causes build failures on OS X 10.6. For more details, see
  https://mta.openssl.org/pipermail/openssl-dev/2016-March/005605.html

Upstream-Status: Submitted [https://mta.openssl.org/pipermail/openssl-dev/2016-March/005605.html]
---
 crypto/perlasm/x86_64-xlate.pl | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/crypto/perlasm/x86_64-xlate.pl b/crypto/perlasm/x86_64-xlate.pl
index 0a023fb..29f94b5 100755
--- ./crypto/perlasm/x86_64-xlate.pl
+++ ./crypto/perlasm/x86_64-xlate.pl
@@ -198,11 +198,8 @@ my %globals;
 	if ($gas) {
 	    # Solaris /usr/ccs/bin/as can't handle multiplications
 	    # in $self->{value}
-	    my $value = $self->{value};
-	    $value =~ s/(?<![\w\$\.])(0x?[0-9a-f]+)/oct($1)/egi;
-	    if ($value =~ s/([0-9]+\s*[\*\/\%]\s*[0-9]+)/eval($1)/eg) {
-		$self->{value} = $value;
-	    }
+	    $self->{value} =~ s/(?<![\w\$\.])(0x?[0-9a-f]+)/oct($1)/egi;
+	    $self->{value} =~ s/([0-9]+\s*[\*\/\%]\s*[0-9]+)/eval($1)/eg;
 	    sprintf "\$%s",$self->{value};
 	} else {
 	    $self->{value} =~ s/(0b[0-1]+)/oct($1)/eig;
-- 
2.7.2

